docs/src/prototypes/chatMessages/ThreadedMessages/ThreadReplyEditor.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react'
+import * as keyboardKey from 'keyboard-key'
+
+import { Button, Flex, Input, Toolbar, Ref, Chat } from '@stardust-ui/react'
+import { toolbarItems } from './mockData'
+
+interface ThreadReplyEditorState {
+  editMode: boolean
+}
+
+class ThreadReplyEditor extends React.Component<{}, ThreadReplyEditorState> {
```

Comment:
If we started to use functional components, let's continue there

Reply:
I need such things as after state callbacks to focus elements after the state changed or when component did update
```jsx
this.setState({ editMode: true }, () => {
     if (this.inputRef && this.inputRef.current) {
          this.inputRef.current.focus()
       }
 })
```

If you know how easy to achieve this in the functional component with hooks let me know :)

Reply:
Focus handling should occur in Effect phase (`useEffect`/`didMount`/didUpdate`): https://reactjs.org/docs/hooks-effect.html

And it can be easily done in `useEffect`, if I correct get it should be something like this:

```tsx
const buttonRef = React.useRef(null)
const inputRef = React.useRef(null)

const [editMode, setEditMode] = React.useState()

React.useEffect(
  () => {
     if (editMode) {
       _.invoke(this.inputRef.current, 'focus')
     } else {
       _.invoke(this.buttonRef.current, 'focus')
     }
  },
  [editMode]
)
```

Reply:
hmm, cool. I'll try that, thank you!